lib/hexpm_web/controllers/login_controller.ex

```diff
@@ -74,6 +77,20 @@ defmodule HexpmWeb.LoginController do
     )
   end
 
+  defp login(conn, %User{id: user_id, tfa_enabled: true}, password_breached: breached?) do
+    conn
+    |> configure_session(renew: true)
+    |> put_session("two_factor_user_id", user_id)
```

Comment:
An alternative way to do this is create a completely separate session with a completely separate cookie, but there's some overhead there, and not sure there would be much gain.